x-pack/plugins/security_solution/public/common/components/navigation/use_security_solution_navigation/use_navigation_items.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { APP_ID } from '../../../../../common/constants';
+import { track, METRIC_TYPE, TELEMETRY_EVENT } from '../../../lib/telemetry';
+import { getSearch } from '../helpers';
+import { PrimaryNavigationItemsProps } from './types';
+import { useKibana } from '../../../lib/kibana';
+import { SecurityPageName } from '../../../../app/types';
+import { useFormatUrl } from '../../link_to';
+
+export const usePrimaryNavigationItems = ({
+  filters,
+  navTabs,
+  query,
+  savedQuery,
+  selectedTabId,
+  sourcerer,
+  timeline,
+  timerange,
+}: PrimaryNavigationItemsProps) => {
+  const history = useHistory();
+  const { navigateToApp, getUrlForApp } = useKibana().services.application;
+
+  const navItems = Object.values(navTabs).map((tab) => {
+    const { id, href, name, disabled, pageId } = tab;
+    // Rules of hooks prevents hooks from being called in callbacks as the order of the hooks
+    // should be consistent across every render. The primary nav tabs here are from a constants file
+    // not dynamically generated, so their order will remain consistent across renders.
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const { formatUrl } = useFormatUrl(((pageId ?? id) as unknown) as SecurityPageName);
```

Comment:
I'm not super familiar with this code but could we set `pageId` on all the tabs here: https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/app/home/home_navigations.tsx#L21 ? Then we could avoid the casting right?

It might also help clean up a few places where we need to check whether `id` or `pageId` is set.

Reply:
Was actually able to remove this entirely. Kept a lot of logic from the tab navigation that wasn't necessary